js/poloniexfutures.js

```diff
@@ -0,0 +1,920 @@
+        }
+        const iceberg = this.safeValue (params, 'iceberg');
+        if (iceberg) {
+            const visibleSize = this.safeValue (params, 'visibleSize');
+            if (visibleSize === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires a visibleSize parameter for iceberg orders');
+            }
+        }
+        params = this.omit (params, [ 'timeInForce', 'stopPrice', 'triggerPrice' ]); // Time in force only valid for limit orders, exchange error when gtc for market orders
+        const response = await this.privatePostOrders (this.extend (request, params));
+        //
+        //    {
+        //        code: "200000",
+        //        data: {
+        //            orderId: "619717484f1d010001510cde",
+        //        },
+        //    }
+        //
+        const data = this.safeValue (response, 'data', {});
+        return {
+            'id': this.safeString (data, 'orderId'),
+            'clientOrderId': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'lastTradeTimestamp': undefined,
+            'symbol': undefined,
+            'type': undefined,
+            'side': undefined,
+            'price': undefined,
+            'amount': undefined,
+            'cost': undefined,
+            'average': undefined,
+            'filled': undefined,
+            'remaining': undefined,
+            'status': undefined,
+            'fee': undefined,
+            'trades': undefined,
+            'timeInForce': undefined,
+            'postOnly': undefined,
+            'stopPrice': undefined,
+            'info': response,
+        };
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api];
+        const versions = this.safeValue (this.options, 'versions', {});
+        const apiVersions = this.safeValue (versions, api, {});
+        const methodVersions = this.safeValue (apiVersions, method, {});
+        const defaultVersion = this.safeString (methodVersions, path, this.version);
+        const version = this.safeString (params, 'version', defaultVersion);
+        const tail = '/api/' + version + '/' + this.implodeParams (path, params);
+        url += tail;
+        const query = this.omit (params, path);
+        if (api === 'public') {
+            const queryLength = Object.keys (query).length;
+            if (queryLength) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            if (method !== 'GET' && method !== 'HEAD') {
+                body = this.urlencode (query);
+                url += '?' + body;
+            }
+            const now = this.milliseconds ().toString ();
+            let payload = now + method + tail;
```

Comment:
Your `payload` does not follow the spec, it should be something like:
![image](https://user-images.githubusercontent.com/43336371/208722627-6c32fc7b-07f5-47ff-9c46-e684a0722e81.png)

but currently has this format: `1671555347881POST/api/v1/orders'`

You should construct it like this:
(pseudo-code)

```Javascript
payload = method + "\n" +
Access path + "\n" 
+ sorted URL-encoded query containing timestamp

Btw Poloniex has a python SDK you can check how they do the signing here: https://github.com/poloniex/polo-sdk-python/blob/0a58715c45767b794df0a29db5186de8987b81b7/polosdk/rest/request.py#L30